pkg/segment/reader/metrics/tagstree/tagstreereader.go

```diff
@@ -198,7 +213,11 @@ func (attr *AllTagTreeReaders) FindTSIDS(mQuery *structs.MetricsQuery) (*tsidtra
 							err = tracker.BulkAddStarTagsOnly(rawTagValueToTSIDs, initMetricName, tf.TagKey, numValueFiltersNonZero)
 						} else {
 							initMetricName = fmt.Sprintf("%v{", mQuery.MetricName)
-							err = tracker.BulkAddStar(rawTagValueToTSIDs, initMetricName, tf.TagKey, numValueFiltersNonZero)
+							if tf.TagOperator == segutils.Regex {
+								err = tracker.BulkAdd(rawTagValueToTSIDs, mQuery.MetricName, tf.TagKey)
```

Comment:
I am not sure why `BulkAdd` is used here? If you keep `BulkAddStar`, what is happening? I believe `BulkAddStar` should work.

Reply:
If the values are not added in allTSIDs then those ids wont be written. In our case the when * is the input we have selectallseries as true which repeatedly adds tsids to the tracker allTSIDs and then during exit (more is false), only those added tsids are picked up. As we are using regex on specific tag filters selectallseries is false in our case and hence we only have rawTagValueToTSIDs which is yet to be added to the tracker allTSIDs. BulkAdd does this so that is why it is used.